### PR TITLE
api: make storageClassClaim immutable

### PR DESCRIPTION
**Pull request by rewantsoni**
add kubebuilder validations to make storageClassClaim spec immutable

### CODE REVIEW

**Comment by rewantsoni** [comment]
depends on: https://github.com/red-hat-storage/ocs-client-operator/pull/72

File: api/v1alpha1/storageclassclaim_types.go [hunk 1]
```diff
@@ -73,7 +73,9 @@ type StorageClassClaim struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   StorageClassClaimSpec   `json:"spec,omitempty"`
+	// +kubebuilder:validation:Required
```
Comment (Madhu-1) [inline api/v1alpha1/storageclassclaim_types.go:76]:
```suggestion
//+kubebuilder:validation:Required
//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec is immutable"
```

Remove space between `//` and `+`

Comment (Madhu-1) [reply api/v1alpha1/storageclassclaim_types.go:76]: and switch `self` and `oldself` as doing here https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Comment (rewantsoni) [reply api/v1alpha1/storageclassclaim_types.go]: done

File: api/v1alpha1/storageclassclaim_types.go [hunk 2]
```diff
@@ -73,7 +73,9 @@ type StorageClassClaim struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   StorageClassClaimSpec   `json:"spec,omitempty"`
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:XValidation:rule="oldSelf == self",message="spec is immutable"
```
Comment (leelavg) [inline api/v1alpha1/storageclassclaim_types.go:77]: does this rule enforces all fields in the struct as well? like, after setting the spec, does `.spec.storageProfile` also becomes immutable?

Comment (nb-ohad) [reply api/v1alpha1/storageclassclaim_types.go:77]: As part of similar work we did for storage profiles, we observed that it does enforce on the entire structure and sub structures. 

@leelavg If you are satisfied please resolve this thread

Comment (leelavg) [reply api/v1alpha1/storageclassclaim_types.go:77]: Can be closed.

**Comment by rewantsoni** [comment]
rebased the branch against main after https://github.com/red-hat-storage/ocs-client-operator/pull/72 merged

**Review by leelavg — APPROVED** [review]

**Pull request merged**